### DISCUSSION TITLE
TypeError: Object of type Decimal is not JSON serializable.

### DISCUSSION THREAD

**Discussion by p-madduri** [apache/airflow#28018] [Q&A]
https://github.com/apache/airflow/blob/430e930902792fc37cdd2c517783f7dd544fbebf/airflow/providers/amazon/aws/transfers/dynamodb_to_s3.py#L39

if we use the below class at line 39:
def _convert_item_to_json_bytes(item: dict[str, Any]) -> bytes:
    return (json.dumps(item) + "\n").encode("utf-8")

 its throwing below error
TypeError: Object of type Decimal is not JSON serializable. 

can we use 

class DecimalEncoder(json.JSONEncoder):
    def encode(self, obj):
        if isinstance(obj, Mapping):
            return '{' + ', '.join(f'{self.encode(k)}: {self.encode(v)}' for (k, v) in obj.items()) + '}'
        elif isinstance(obj, Iterable) and (not isinstance(obj, str)):
            return '[' + ', '.join(map(self.encode, obj)) + ']'
        elif isinstance(obj, Decimal):
            return f'{obj.normalize():f}'  # using normalize() gets rid of trailing 0s, using ':f' prevents scientific notation
        else:
            print(obj)
        return super().encode(obj)
        
   and update the code

        body = json.dumps(data, cls=DecimalEncoder)
        s3.Object(s3_bucket, s3_object + filename).put(Body=json.dumps(data))

**Comment by Taragolis** [comment]
Feel free to make a pull request: https://github.com/apache/airflow/blob/main/CONTRIBUTORS_QUICK_START.rst

**Reply by p-madduri** [reply]
can someone create the pull request? i can't do it. i am just using that operator.

**Reply by potiuk** [reply]
Anyone. Including you. Airlfow has > 2200 contributors who made pull requests. Most of them were users who decided they want to contribute their code.

**Reply by potiuk** [reply]
It's the least you can do to give back for the absolutely free software you get, I think.

**Reply by potiuk** [reply]
And you if you do not see capable/willing , you can always create an issue and describe it in the way that someone might pick it up. We will mark it as "good first issue" and hopefuly some day someone will pick it up.
